bin/rose-date

```diff
@@ -175,4 +173,34 @@
 #
 #     For example, for a duration `P57DT12H` - `y,m,d,h` -> `0,0,57,12`
 #-------------------------------------------------------------------------------
-exec python3 -m rose.date "$@"
+# -*- coding: utf-8 -*-
+import re
+import sys
+import os
+
+from isodatetime.main import main
+
+if __name__ == '__main__':
+    sys.argv[0] = re.sub(r'(-script\.pyw?|\.exe)?$', '', sys.argv[0])
+
+    # Handle Legacy Rose-date -c functionality
+    if '-c' in sys.argv or '--use-task-cycle-time' in sys.argv:
+        rose_task_cycle_time = os.getenv('ROSE_TASK_CYCLE_TIME')
+        isodatetimeref = os.getenv('ISODATETIMEREF')
+        if rose_task_cycle_time:
+            os.environ['ISODATETIMEREF'] = rose_task_cycle_time
+        elif isodatetimeref:
+            pass
+        else:
+            sys.exit("[FAIL] [UNDEFINED ENVIRONMENT VARIABLE] ROSE_TASK_CYCLE_TIME")
```

Comment:
Can probably replace the `elif ... else` with just `elif not isodatetimeref:`?

Reply:
Or even `elif not os.getenv('ISODATETIMEREF'):` and avoid creating another variable `isodatetimeref`?

Reply:
The `elif not` logic looks cool. I orginally wrote this without creating the internal variables, I thought it looked nicer. Now I'm not so sure.